assignment/scripts/array-practice.js

```diff
@@ -75,20 +78,48 @@ console.log('The animals are now', animalArray);
 
 // 4.d TODO: Remove the food at the beginning of your array & 
 //     log both the food removed and the updated array
+removedFood = favoriteFoods.shift();
+console.log('Removed the first food', removedFood);
+console.log('The foods are now', favoriteFoods);
 
 // 4.e (STRETCH) TODO: Replace the second food in your array
 //      with another one of your favorite foods.
 //      Then log the updated array.
 
+/* found appropriate usage of .splice here:
+https://stackoverflow.com/a/586189
+=> modified to replace rather than insert
+ */
+favoriteFoods.splice(
+    1, // the index at which to splice
+    1, // number of elements to remove
+    'italian bread' // first element to insert
+)
+console.log('Replaced bread type', favoriteFoods);
+
 // 4.f (STRETCH) TODO: Sort your favoriteFoods array
 //     in reverse alphabetical order. Log the array.
+favoriteFoods.sort();
```

Comment:
Close, but the prompt specified _reverse_ alphabetical order. The default `sort` sort is alphabetical.